server/routes/subscription/createSubscription.ts

```diff
@@ -26,8 +26,8 @@ export default async (models, req: Request, res: Response, next: NextFunction) =
   const p_entity = parsed_object_id[0];
   let chain;
 
-  switch (category.name) {
-    case 'new-thread-creation': {
+  switch (true) {
+    case (category.name === 'new-thread-creation'): {
```

Comment:
why invert the switch statement to work like this?

it seems logically okay but is confusing and more brittle (if the switch ever gets changed so two statements could both be true it will malfunction)

why not just have two cases that collapse into each other?

```

switch (x)
case 1:
case 2:
  break;
case 3:
  // etc...
```

Reply:
great suggestion, i totally forgot about Fallthroughs